### PR TITLE
Activating widget also if GLSP widget is a child

### PR DESCRIPTION
**Pull request by planger**
Fixes https://github.com/eclipse-glsp/glsp/issues/168

### CODE REVIEW

File: packages/theia-integration/src/browser/theia-model-source-changed-handler.ts [hunk 1]
```diff
@@ -30,22 +30,23 @@ export class TheiaModelSourceChangedHandler extends ExternalModelSourceChangedHa
...
+                return this.notifyModelSourceChangedWithWidget(widget, modelSourceName);
             }
         }
         return [];
     }
 
-    protected async notifyModelSourceChangedWithWidget(diagramWidget: GLSPDiagramWidget, modelSourceName: string): Promise<Action[]> {
+    protected async notifyModelSourceChangedWithWidget(widget: Widget, modelSourceName: string): Promise<Action[]> {
+        const diagramWidget = this.getDiagramWidget(widget);
+        if (!(diagramWidget instanceof GLSPDiagramWidget)) {
```
Comment (tortmayr) [inline packages/theia-integration/src/browser/theia-model-source-changed-handler.ts:41]: `this.getDiagramWidget()`  returns `GLSPDiagramWidget|undefined` so there is no need to typecheck here.
`if (!diagramWidget)` should be enough

**Review by tortmayr — APPROVED** [review]
Looks good to me. I noticed one unnecessary typecheck, but that should not prevent merging
:+1:

**Comment by planger** [comment]
@martin-fleck-at Regarding our discussion we had offline: we can already customize the actions that are sent on reload by overwriting the diagram widget (reloadModel()).

**Comment by planger** [comment]
Thanks for the review @tortmayr!

**Pull request merged**